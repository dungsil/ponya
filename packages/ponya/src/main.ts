import { createApp } from 'vue-demi'
import App from './app.vue'

import 'uno.css'

createApp(App)
  .mount('#top')
