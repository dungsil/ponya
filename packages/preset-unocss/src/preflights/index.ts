import type { Preflight } from '@unocss/core'
import { Theme } from '@unocss/preset-mini'

// noinspection CssUnknownProperty
export const preflights: Preflight<Theme>[] = [
  {
    layer: 'preflights',
    // language=CSS
    getCSS: ({ theme }) => `
        :root {
            
        }
    `
  }
]
