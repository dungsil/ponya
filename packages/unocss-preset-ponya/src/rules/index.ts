import type { Rule } from '@unocss/core'
import type { Theme } from '../theme'
import { wh, minOrMaxWh, minMaxWh, mb, mbXy, display, flexDirections } from './layouts'


export const rules: Rule<Theme>[] = [
  // Layouts
  display,
  flexDirections,

  wh, minOrMaxWh, minMaxWh,
  mb, mbXy,

  // Typography
]
