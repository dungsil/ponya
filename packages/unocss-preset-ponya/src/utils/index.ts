export * from './css'
export * from './colors'
