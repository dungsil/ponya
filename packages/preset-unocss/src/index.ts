import { type Preset, Rule } from 'unocss'
import { type Theme } from '@unocss/preset-mini'
import { rules } from '@unocss/preset-mini/rules'

import { theme } from './theme'

/**
 * Ponya preset
 */
export function presetPonya(): Preset<Theme> {


  return {
    name: '@ponya-ui/preset-unocss',
    theme,
    rules: [
      // base rules
      ...rules as Rule<Theme>[]
    ],
  }
}
