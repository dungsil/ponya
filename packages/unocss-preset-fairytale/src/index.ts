import { type Preset } from 'unocss'

import { theme, type Theme } from './theme'
import { preflights } from './preflights'
import { rules } from './rules'

/**
 * Fairytale preset
 */
export function presetFairytale(): Preset<Theme> {

  return {
    name: 'unocss-preset-fairytale',
    theme,
    preflights,
    rules,
  }
}
