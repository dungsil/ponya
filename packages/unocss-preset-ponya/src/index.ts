import { type Preset } from 'unocss'

import { theme, type Theme } from './theme'
import { preflights } from './preflights'
import { rules } from './rules'

/**
 * Ponya preset
 */
export function presetPonya(): Preset<Theme> {

  return {
    name: 'unocss-preset-ponya',
    theme,
    preflights,
    rules,
  }
}
