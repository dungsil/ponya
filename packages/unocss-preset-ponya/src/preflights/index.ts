import type { Preflight } from '@unocss/core'
import type { Theme } from '../theme'
import { colorResolver } from '../utils'

// noinspection CssUnknownProperty
export const preflights: Preflight<Theme>[] = [
  {
    layer: 'preflights',
    // language=CSS
    getCSS: ({ theme }) =>  `
html[data-color-scheme="auto"] {
    color-scheme: normal;    
}
html[data-color-scheme="light"] {
    color-scheme: light;
}
html[data-color-scheme="dark"] {
    color-scheme: dark;
}

body {
  font-family: ${theme.font.family[theme.base.font.family]};
  font-weight: ${theme.font.weight[theme.base.font.weight]};
  font-size: ${theme.font.size[theme.base.font.size]};
  line-height: ${theme.lineHeight[theme.base.lineHeight]};
  letter-spacing: ${theme.letterSpacing[theme.base.letterSpacing]};
  color: ${colorResolver(theme, theme.base.font.color)};
}
    `
  }
]
