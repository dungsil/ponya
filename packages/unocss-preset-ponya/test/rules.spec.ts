import { describe, test, expect, it } from 'vitest'
import { createGenerator } from 'unocss'
import { presetPonya } from '../src'

const uno = createGenerator({
  // @ts-ignore
  presets: [presetPonya()]
})

describe('Layouts', () => {
  // wh
  test('width - length', async () => await matches(['w1', 'w100p', 'w100px']))
  test('width - keyword', async () => await matches(['w-none', 'w-auto', 'w-inherit', 'w-initial']))
  test('height - length', async () => await matches(['h1', 'h100p', 'h100vh']))
  test('height - keyword', async () => await matches(['h-none', 'h-auto', 'h-inherit', 'h-initial']))

  // wh - minOrMax
  test('min-width - length', async () => await matches(['w~1', 'w~10rem', 'w~100p']))
  test('min-width - keyword', async () => await matches(['w~auto', 'w~none', 'w~inherit', 'w~initial']))
  test('max-width - length', async () => await matches(['w1~', 'w10rem~', 'w100p~']))
  test('max-width - keyword', async () => await matches(['w-auto~', 'w-none~', 'w-inherit~', 'w-initial~']))
  test('min-height - length', async () => await matches(['h~1', 'h~10rem', 'h~100p']))
  test('min-height - keyword', async () => await matches(['h~auto', 'h~none', 'h~inherit', 'h~initial']))
  test('max-height - length', async () => await matches(['h1~', 'h10rem~', 'h100p~']))
  test('max-height - keyword', async () => await matches(['h-auto~', 'h-none~', 'h-inherit~', 'h-initial~']))

  // wb - minAndMax
  test('min/max-width', async () => await matches(['w1~2', 'w10p~20', 'w10~20p', 'w1rem~100p']))
  test('min/max-height', async () => await matches(['h1~2', 'h10p~20', 'h10~20p', 'h1rem~100p']))

  // mb
  test('margin', async () => await matches(['m1', 'm10p', 'm10px', 'ml-auto', 'mb1vh', 'mt2', 'mr-none', 'mx-auto']))
  test('padding', async () => await matches(['p1', 'p10p', 'p10px', 'pl-auto', 'pb1vh', 'pt2', 'pr-none', 'py0']))
})

async function matches(selectors: string[]) {
  const { matched, css } = await uno.generate(selectors.join(' '), { preflights: false })

  expect(Array.from(matched)).toEqual(selectors)
  expect(css).toMatchSnapshot()
}
