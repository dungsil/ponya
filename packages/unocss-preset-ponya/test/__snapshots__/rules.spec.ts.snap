// Vitest Snapshot v1

exports[`Layouts > height - keyword 1`] = `
"/* layer: default */
.h-auto{height:auto;}
.h-inherit{height:inherit;}
.h-initial{height:initial;}
.h-none{height:none;}"
`;

exports[`Layouts > height - length 1`] = `
"/* layer: default */
.h1{height:4px;}
.h100p{height:100%;}
.h100vh{height:100vh;}"
`;

exports[`Layouts > margin 1`] = `
"/* layer: default */
.m1{margin:4px;}
.m10p{margin:10%;}
.m10px{margin:10px;}
.mb1vh{margin-bottom:1vh;}
.ml-auto{margin-left:auto;}
.mr-none{margin-right:none;}
.mt2{margin-top:8px;}
.mx-auto{margin-left:auto;margin-right:auto;}"
`;

exports[`Layouts > max-height - keyword 1`] = `
"/* layer: default */
.h-auto\\\\~{min-height:auto;}
.h-inherit\\\\~{min-height:inherit;}
.h-initial\\\\~{min-height:initial;}
.h-none\\\\~{min-height:none;}"
`;

exports[`Layouts > max-height - length 1`] = `
"/* layer: default */
.h1\\\\~{min-height:4px;}
.h100p\\\\~{min-height:100%;}
.h10rem\\\\~{min-height:10rem;}"
`;

exports[`Layouts > max-width - keyword 1`] = `
"/* layer: default */
.w-auto\\\\~{min-width:auto;}
.w-inherit\\\\~{min-width:inherit;}
.w-initial\\\\~{min-width:initial;}
.w-none\\\\~{min-width:none;}"
`;

exports[`Layouts > max-width - length 1`] = `
"/* layer: default */
.w1\\\\~{min-width:4px;}
.w100p\\\\~{min-width:100%;}
.w10rem\\\\~{min-width:10rem;}"
`;

exports[`Layouts > min/max-height 1`] = `
"/* layer: default */
.h1\\\\~2{min-height:4px;max-height:8px;}
.h10\\\\~20p{min-height:40px;max-height:20%;}
.h10p\\\\~20{min-height:10%;max-height:80px;}
.h1rem\\\\~100p{min-height:1rem;max-height:100%;}"
`;

exports[`Layouts > min/max-width 1`] = `
"/* layer: default */
.w1\\\\~2{min-width:4px;max-width:8px;}
.w10\\\\~20p{min-width:40px;max-width:20%;}
.w10p\\\\~20{min-width:10%;max-width:80px;}
.w1rem\\\\~100p{min-width:1rem;max-width:100%;}"
`;

exports[`Layouts > min-height - keyword 1`] = `
"/* layer: default */
.h\\\\~auto{max-height:auto;}
.h\\\\~inherit{max-height:inherit;}
.h\\\\~initial{max-height:initial;}
.h\\\\~none{max-height:none;}"
`;

exports[`Layouts > min-height - length 1`] = `
"/* layer: default */
.h\\\\~1{max-height:4px;}
.h\\\\~100p{max-height:100%;}
.h\\\\~10rem{max-height:10rem;}"
`;

exports[`Layouts > min-width - keyword 1`] = `
"/* layer: default */
.w\\\\~auto{max-width:auto;}
.w\\\\~inherit{max-width:inherit;}
.w\\\\~initial{max-width:initial;}
.w\\\\~none{max-width:none;}"
`;

exports[`Layouts > min-width - length 1`] = `
"/* layer: default */
.w\\\\~1{max-width:4px;}
.w\\\\~100p{max-width:100%;}
.w\\\\~10rem{max-width:10rem;}"
`;

exports[`Layouts > padding 1`] = `
"/* layer: default */
.p1{padding:4px;}
.p10p{padding:10%;}
.p10px{padding:10px;}
.pb1vh{padding-bottom:1vh;}
.pl-auto{padding-left:auto;}
.pr-none{padding-right:none;}
.pt2{padding-top:8px;}
.py0{padding-top:0;padding-bottom:0;}"
`;

exports[`Layouts > width - keyword 1`] = `
"/* layer: default */
.w-auto{width:auto;}
.w-inherit{width:inherit;}
.w-initial{width:initial;}
.w-none{width:none;}"
`;

exports[`Layouts > width - length 1`] = `
"/* layer: default */
.w1{width:4px;}
.w100p{width:100%;}
.w100px{width:100px;}"
`;
